hbase-common/src/main/java/org/apache/hadoop/hbase/ServerName.java

```diff
@@ -356,13 +359,14 @@ public static ServerName parseServerName(final String str) {
    * Checks if the provided server address is formatted as an IPv6 address. This method uses Java's
    * InetAddress to parse the input and confirms if the address is an instance of Inet6Address for
    * robust validation instead of relying on string splitting.
-   * @param serverAddress The address string to validate (can be a hostname or IP)
+   * @param serverAddress The address string to validate
    * @return true if the address is a valid IPv6 address; false otherwise.
    */
   public static boolean isIpv6ServerName(String serverAddress) {
     try {
       InetAddress address = InetAddress.getByName(serverAddress);
-      return address instanceof Inet6Address;
+      return address instanceof Inet6Address && !address.getHostName().equals(serverAddress)
```

Comment:
`!address.getHostName().equals(serverAddress)`

This excludes the cases when the canonical IPv6 address format is supplied.
Why would we want to do that ?

The colon check looks good.